File: native-integration/react-native-expo/MintClient.ts
import { Config } from './config';

/**
 * Returns a one-time `embedToken` for AUTO login. Manual login needs none of this.
 *
 * PRODUCTION: your backend mints the token (the `aiq_embed_` secret stays server-side) and you POST your
 * own mint endpoint (`Config.MINT_ENDPOINT`). For a quick LOCAL test, set `Config.TEST_EMBED_SECRET` and the
 * app mints directly against the portal — never ship a real secret in an app (it's extractable from the build).
 */
export async function getEmbedToken(): Promise<string> {
  if (Config.TEST_EMBED_SECRET) {
    const r = await fetch(`${Config.BASE_URL}/api/v1/embed/sessions`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${Config.TEST_EMBED_SECRET}`,
      },
      body: JSON.stringify({ employee: { email: Config.DEMO_EMAIL }, autoProvision: true }),
    });
    const j = await r.json();
    return j?.data?.embedToken ?? j?.embedToken;
  }
  const r = await fetch(Config.MINT_ENDPOINT, { method: 'POST' });
  const j = await r.json();
  return j.embedToken;
}

File: native-integration/flutter/lib/mint_client.dart
import 'dart:convert';

import 'package:http/http.dart' as http;

import 'config.dart';

/// Returns a one-time `embedToken` for AUTO login. Manual login needs none of this.
///
/// PRODUCTION: your backend mints the token (the `aiq_embed_` secret stays server-side) and you POST
/// your own mint endpoint (`Config.mintEndpoint`). For a quick LOCAL test, set `Config.testEmbedSecret`
/// and the app mints directly against the portal — never ship a real secret in an app.
Future<String> getEmbedToken() async {
  if (Config.testEmbedSecret.isNotEmpty) {
    final r = await http.post(
      Uri.parse('${Config.baseUrl}/api/v1/embed/sessions'),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer ${Config.testEmbedSecret}',
      },
      body: jsonEncode({
        'employee': {'email': Config.demoEmail},
        'autoProvision': true,
      }),
    );
    final j = jsonDecode(r.body) as Map<String, dynamic>;
    final data = j['data'] as Map<String, dynamic>?;
    final token = (data?['embedToken'] ?? j['embedToken']) as String?;
    if (token == null) throw Exception('mint failed: ${r.statusCode} ${r.body}');
    return token;
  }
  final r = await http.post(Uri.parse(Config.mintEndpoint));
  final j = jsonDecode(r.body) as Map<String, dynamic>;
  final token = j['embedToken'] as String?;
  if (token == null) throw Exception('mint failed: ${r.statusCode}');
  return token;
}
